crates/libp2p-networking/src/network/node/handle.rs

```diff
@@ -294,12 +294,7 @@ impl<S> NetworkNodeHandle<S> {
         dht_timeout: Duration,
     ) -> Result<PeerId, NetworkNodeHandleError> {
         // get record (from DHT)
-        let pid = self.get_record_timeout::<PeerId>(&key, dht_timeout).await?;
-
-        // pid lookup for routing
-        self.lookup_pid(pid).await?;
-
-        Ok(pid)
+        self.get_record_timeout::<PeerId>(&key, dht_timeout).await
```

Comment:
I removed it because I found it seemed more efficient to perform when we needed it, at least in the case nodes were down. It was exacerbating the issue where we were doing infinite retries; I will see if we can add it back.

Reply:
Was able to add this back !